src/compiler/program.ts

```diff
@@ -247,6 +251,13 @@ module ts {
             return bindDiagnostics.concat(checkDiagnostics).concat(programDiagnostics);
         }
 
+        function getDeclarationDiagnosticsForFile(sourceFile: SourceFile): Diagnostic[] {
+            if (!isDeclarationFile(sourceFile)) {
+                let resolver = getDiagnosticsProducingTypeChecker().getEmitResolver(sourceFile);
+                return ts.getDeclarationDiagnostics(getEmitHost(), resolver, sourceFile);

```

Comment:
Pass an empty writeFile callback so there are no side effects, i.e. emitting the files.